app/src/main/java/com/github/polypoly/app/map/MapActivity.kt

```diff
@@ -136,6 +144,73 @@ class MapActivity : ComponentActivity() {
         }, modifier = Modifier.testTag("map"))
     }
 
+    @Composable
+    fun RollDiceButton() {
+        Box(modifier = Modifier.fillMaxWidth()) {
+            Button(
+                modifier = Modifier
+                    .size(80.dp)
+                    .align(Alignment.BottomCenter)
+                    .offset(y = (-80).dp)
+                    .testTag("rollDiceButton"),
+                onClick = {
+                    // Toast.makeText(this@MapActivity, rollDice(), Toast.LENGTH_SHORT).show()
+                    showRollDiceDialog.value = true
+                },
+                shape = CircleShape
+
+            ) {
+                Icon(Icons.Filled.Casino, contentDescription = "Roll Dice")
+            }
+        }
+    }
+
+    /**
+     * Dice roll dialog, shows the result of 3 dice rolls in a column.
+     */
+    @Composable
+    fun RollDiceDialog() {
+        if (showRollDiceDialog.value) {
+            Dialog(onDismissRequest = { showRollDiceDialog.value = false }) {
+                AlertDialog(
+                    onDismissRequest = { showRollDiceDialog.value = false },
+                    title = { Text("Dice Roll") },
+                    text = {
+                        Column {
+                            // 3 buttons, containing the name of a random location
+                            Button(onClick = { }) {
+                                Text(rollDice().name)
+                            }
+                            Button(onClick = { }) {
+                                Text(rollDice().name)
+                            }
+                            Button(onClick = { }) {
+                                Text(rollDice().name)
+                            }
+                        }
+                    },
+                    confirmButton = {
+                        Button(onClick = { showRollDiceDialog.value = false }) {
+                            Text("Quit")
+                        }
+                    }
+                )
+            }
+        }
+    }
+
+    private fun rollDice(): com.github.polypoly.app.game.Location {
+        val diceRollsSum = IntArray(2) { (1..6).random() }.sum() - 1
+
+        // get the 12 closest locations
```

Comment:
I would find it better that with each roll of the dice, any place can fall and not the nearest ones. The idea is that when you get a 6 you have to go to the other end of campus. So I suggest this algorithm:
- Sort locations by distance, from closest to least close.
- Remove the one or those who visited the previous round
- Divide the total number of locations by 11 (`n = nbLocations/11`)
- Compute the rest ( `r = nbLocations%11` )
- Draw two random numbers between 1 and 6 and remove two (`diceRollsSum = IntArray(2) { (1..6).random() }.sum() - 2`)
- Select three random locations with an index between `diceRollsSum*(n+1)` and `(diceRollsSum + 1)*(n+1) - 1` if `diceRollsSum < r`, else between `r*(n+1) + (diceRollsSum - r)*n` and `r*(n+1) + (diceRollsSum - r +1)*n -1`
- if there are less than 3 locations in this gap, then take 1 locations before or after

Reply:
Requires more info about the game, such as previously visited locations, and not sure to agree. To be discussed.
Cannot be implemented yet, read.

Reply:
ok ok no problem!